pkg/configuration/component.go

```diff
@@ -0,0 +1,46 @@
+package configuration
+
+import (
+	"context"
+
+	errorhandler "github.com/cloudtrust/common-service/v2/errors"
+	"github.com/cloudtrust/common-service/v2/log"
+	"github.com/cloudtrust/keycloak-bridge/internal/keycloakb"
+)
+
+type component struct {
+	contextKeyMgr ContextKeyManager
+	logger        log.Logger
+}
+
+type Component interface {
+	GetIdentificationURI(ctx context.Context, realm string, contextKey string) (string, error)
+}
+
+// ContextKeyManager interface
+type ContextKeyManager interface {
+	GetOverride(realm string, contextKey string) (keycloakb.ContextKeyParameters, bool)
+}
+
+func NewComponent(contextKeyManager ContextKeyManager, logger log.Logger) Component {
+	return &component{
+		contextKeyMgr: contextKeyManager,
+		logger:        logger,
+	}
+}
+
+// GetIdentificationURI returns the identification URI for a given context key and realm
+func (c *component) GetIdentificationURI(ctx context.Context, realm string, contextKey string) (string, error) {
+	ctxOverride, ok := c.contextKeyMgr.GetOverride(realm, contextKey)
+	if !ok {
+		c.logger.Info(ctx, "msg", "Invalid (context-key, realm) pair", "context-key", contextKey, "realm", realm)
+		return "", errorhandler.CreateBadRequestError(errorhandler.MsgErrInvalidParam + ".context-key")
+	}
+
+	if ctxOverride.IdentificationURI == nil {
+		c.logger.Info(ctx, "msg", "Empty identification URI", "context-key", contextKey, "realm", realm)
+		return "", errorhandler.CreateBadRequestError(errorhandler.MsgErrInvalidParam + ".context-key")
```

Comment:
I'm not sure about those bad request errors, so don't hesitate to suggest other ideas.

Reply:
It's not a context-key related error... This could be a 5xx error as there is a problem on our side because realm is not correctly configured.
400 Bad request => invalid-param.no-uri?
412 Precondition Failed? Precondition would be realm should be configured first? => precondition.realm-configuration
417 Expectation failed? (almost the same as 412) => expectation.realm-configuration

Reply:
Another way of solving this question, at startup, if identification URI is not provided, make the service start fails... (are there cases where we don't want to configure it?)

Reply:
I'll look into it, thanks for the suggestions

Reply:
I decided to go with 404 to emphasize that we do not have an identification URI to provide for this (realm, context-key) pair

Reply:
I would not make this service fails at startup for bad context-key configuration. I think it is kind of not flexible, we want maybe configure at the end for all customer. But maybe not at the same time.

For me it is an error part of 500 errors family